### PR TITLE
MAT-5434: Adding some additional models to help with the creation of …

### PR DESCRIPTION
**Pull request by gregory-akins**
…Coding elements in the output

## MADiE FHIR SERVICE

Jira Ticket: [MAT-5434](https://jira.cms.gov/browse/MAT-5434)
(Optional) Related Tickets:

### Summary

The codeFilter element on the exported DataRequirements was missing a "coding" atrribute.  Adding additional models to support that.

### All Submissions
* [ X] This PR has the JIRA linked.
* [X] Required tests are included.
* [X] No extemporaneous files are included (i.e Complied files or testing results).
* [X] This PR is merging into the **correct branch**.
* [X] All Documentation needed for this PR is Complete (or noted in a TODO or other Ticket).
* [ ] Any breaking changes or failing automations are noted by placing a comment on this PR.

### DevSecOps
If there is a question if this PR has a security or infrastructure impact, please contact the Security or DevOps engineer assigned to this project to discuss it further.

* [ ] This PR has NO significant security impact (i.e Changing auth methods, Adding a new user type, Adding a required but vulnerable package).

### Reviewers
By Approving this PR you are attesting to the following:

*  Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose.
*  The tests appropriately test the new code, including edge cases.
*  If you have any concerns they are brought up either to the developer assigned, security engineer, or leads.

### CODE REVIEW

**Comment by nmorasb** [comment]
seems like we don't have any unit tests for the LibraryCqlVisitor. With all the changes to this class, is it time to introduce a unit test file for that class?
Also I think we're missing the build checks for Codacy and codecov as I feel like those would have caught the missing test coverage.

**Review by adongare — APPROVED** [review]

File: src/main/java/gov/cms/madie/madiefhirservice/cql/LibraryCqlVisitor.java [hunk 1]
```diff
@@ -198,10 +242,89 @@ private void handleDataRequirement(String type, String path, String valueSetOrCo
...
+      } else {
+        log.debug("Could not find valueset with name " + valueSetName);
+      }
+    } /*
+       * else { // Check to see if its a lib reference like TJC."value set id" int periodIndex =
+       * valueSetName.indexOf("."); if (periodIndex != -1) { String alias =
+       * valueSetName.substring(0, periodIndex); String remaining =
+       * trimQuotes(valueSetName.substring(periodIndex + 1)); var childLib = libMap.get(alias); if
+       * (childLib != null) { result = childLib.getRight().getValueSetUrl(remaining); } } }
+       */
```
Comment (jkotanchik-SB) [inline src/main/java/gov/cms/madie/madiefhirservice/cql/LibraryCqlVisitor.java:302]: Is this an eventual TODO or can it be removed?

**Review by jkotanchik-SB — APPROVED** [review]
